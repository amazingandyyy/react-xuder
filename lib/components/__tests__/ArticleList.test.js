import React from 'react';
import ArticleList from '../ArticleList';

import Enzyme from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
Enzyme.configure({ adapter: new Adapter() });
const {shallow} = Enzyme;

describe('ArticleList', () => {

  const testProps = {
    articles: {
      a: { id: 'a', title: 'title a'},
      b: { id: 'b', title: 'title b'},
    }
  };

  it('renders correctly', () => {
    const wrapper = shallow(
      <ArticleList
        {...testProps}
      />
    ).getElement();
    expect(wrapper.props.children.length).toBe(2);
    expect(wrapper).toMatchSnapshot();
  });
});