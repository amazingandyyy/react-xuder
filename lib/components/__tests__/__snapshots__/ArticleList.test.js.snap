// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ArticleList renders correctly 1`] = `
<div>
  <withStore
    article={
      Object {
        "id": "a",
        "title": "title a",
      }
    }
  />
  <withStore
    article={
      Object {
        "id": "b",
        "title": "title b",
      }
    }
  />
</div>
`;
